model/map/src/main/java/org/keycloak/models/map/realm/MapRealmAdapter.java

```diff
@@ -848,7 +862,7 @@ public void updateAuthenticatorConfig(AuthenticatorConfigModel model) {
         entity.getAuthenticatorConfig(model.getId())
                         .ifPresent(oldAC -> {
                             oldAC.setAlias(model.getAlias());
-                            oldAC.setConfig(model.getConfig());
+                            oldAC.setConfig(model.getConfig() == null ? null : new HashMap<>(model.getConfig()));
```

Comment:
This shouldn't be necessary. Each implementation of `setConfig` should make a deep copy of the parameter. 

Current implementation looks like this:
```java
@SuppressWarnings("unchecked") @Override public void setConfig(java.util.Map<String,String> p0) {
        p0 = p0 == null ? null : new java.util.HashMap<>(p0);
        if (p0 != null) {
            p0.values().removeIf(org.keycloak.models.map.common.UndefinedValuesUtils::isUndefined);
            if (org.keycloak.models.map.common.UndefinedValuesUtils.isUndefined(p0)) p0 = null;
        }
        updated |= ! Objects.equals(fConfig, p0);
        fConfig = p0;
    }
```